src/gcp/cloudscheduler.ts

```diff
@@ -42,8 +49,13 @@ export async function createOrReplaceJob(schedule: Schedule): Promise<void> {
     if (e.context.response.statusCode !== 409) {
       throw e;
     }
-    logLabeledBullet("functions", `re-creating scheduler job ${jobName}`);
-    await deleteJob(schedule.name);
-    return createJob(schedule);
+    const existingJob = await getJob(schedule.name);
+    if (existingJob.body.schedule !== schedule.schedule) {
```

Comment:
We need to compare more than this. If any of the parameters that can be specified are different (e.g. time zone, retry policy, etc) we would still need to recreate.

Reply:
Also what happens if there isn't already a scheduled job? Won't this just crash looking for `existingJob.body`? Make sure this handles the case where there isn't an existing job.

Reply:
To the second comment, the case where there is no existing job is covered by the try block above. However, now that I look at this, i think it can be refactored to be a bit clearer and to not rely on a createJob call erroring out.